starknet-core/src/lib.rs

```diff
@@ -1,6 +1,19 @@
+#![cfg_attr(not(feature = "std"), no_std)]
 #![allow(clippy::comparison_chain)]
 #![doc = include_str!("../README.md")]
 
+#[cfg(feature = "std")]
+include!("./with_std.rs");
+
+#[cfg(not(feature = "std"))]
+include!("./without_std.rs");
+
+pub mod stdlib {
+    #[cfg(feature = "std")]
+    pub use crate::with_std::*;
+    #[cfg(not(feature = "std"))]
+    pub use crate::without_std::*;
+}
```

Comment:
As the rest of the repo doesn't use this method to support no_std, I would suggest continuing doing it the same way: at the file level, by importing from core and alloc rather than std + have hashbrown as a dependency